### PR TITLE
EOS-22649: Node stop REST API implementation.

### PR DESCRIPTION
**Pull request by shrimetta**
# Backend

# Problem Statement

- _Overview: JIRA number/GitHub Issue added to PR: [Y/N]:_
- _Describe the problem this patch intends to solve._

## Design

- _For Bug describe the fix here._
- _For feature, post the link to the solution page._
- _Any northbound interface change and has been notified to other gatekeepers [Y/N]:_

## Coding

- _Coding conventions are followed and code is consistent. Yes/No?_
- _Confirm All CODACY errors are resolved. Yes/No?_

## Testing

- _Confirm that Test Cases are added (for both the cases, fix and feature). Yes/No?_
- _Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability. Yes/No?_
- _Confirm Testing was performed with installed RPM. Yes/No?_
- _Confirm all the ut/regression/smoke test has been performed [Y/N]_

## Checklist

- _PR is self-reviewed? Yes/No?_
- _GitHub Issue is updated_
- _Jira is updated_
  -  _Check if the description is clear and explained._ 
    -  _Check Acceptance Criterion is defined._
      -  _All the tests performed should be mentioned before Resolving a JIRA._
        -  _Verification needs to be done before marked as Closed/Verified_
        - _Any interface change Yes/No?_
        - _Change notified to other gatekeepers: Yes/No?_
        - _Code reviews done and addressed: Yes/No?_
        - _Codacy issues reviewed and addressed: Yes/No?_
        - _Deployment test : Done/Not?_
        - _UT/smoke test: Done/Not?_
        - _Jira number added to PR: Yes/No?_
        - _Github merge done using UI: Yes/No?_
        - _Side effects on other features - deployment/update/node-replacement_
        - _Changes to quick-start-guide/community impact_
        - _All dependent component code PR are raised or already merged Yes/No_
        - _All dependent code already merged in landing branch Yes/No_

Signed-off-by: Shri Metta <shri.metta@seagate.com>

### CODE REVIEW

File: csm/common/ha/cluster_management/operations.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+    def execute(self, cluster_manager, **kwargs):
+        node_id = kwargs.get(const.ARG_RESOURCE_ID, "")
+        args = {
+            const.ARG_CHECK_CLUSTER: not kwargs.get(const.ARG_FORCE, False),
+            const.ARG_POWER_OFF: True,
+            const.ARG_STORAGE_OFF: kwargs.get(const.ARG_STORAGE_OFF, False)
+        }
+        Log.debug(f"NodePoweroffOperation on node with id {node_id} and args: {args}")
+        # try:
```
Comment (pawankumarsrivastava) [inline csm/common/ha/cluster_management/operations.py:134]: Remove the commented lines. We will merge along with HA

File: csm/common/ha/cluster_management/operations.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        errors = ValidatorSchema().validate(kwargs)
+        self.parse_errors(errors)
+
+    def execute(self, cluster_manager, **kwargs):
+        node_id = kwargs.get(const.ARG_RESOURCE_ID, "")
+        args = {
+            const.ARG_CHECK_CLUSTER: not kwargs.get(const.ARG_FORCE, False)
+        }
+        Log.debug(f"NodeStopOperation on node with id {node_id} and args: {args}")
+        # try:
```
Comment (pawankumarsrivastava) [inline csm/common/ha/cluster_management/operations.py:105]: Remove the commented lines. We will merge along with HA

File: csm/common/ha/cluster_management/operations.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+import asyncio
+from concurrent.futures import ThreadPoolExecutor
+from functools import partial
+from abc import ABC, abstractmethod
+
```
Comment (pawankumarsrivastava) [inline csm/common/ha/cluster_management/operations.py:20]: Remove un-necessary blank line.

File: csm/common/ha/cluster_management/operations.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+from marshmallow import Schema, fields, validate
+
+from csm.core.blogic import const
+from csm.common.errors import InvalidRequest
+from cortx.utils.log import Log
+
+
+class Operation(ABC):
+
```
Comment (pawankumarsrivastava) [inline csm/common/ha/cluster_management/operations.py:29]: Add comments for the classes

**Review by pawankumarsrivastava — CHANGES_REQUESTED** [review]
Remove WIP

File: csm/common/ha/cluster_management/operations.py (hunk 1 above)
Comment (shrimetta) [reply csm/common/ha/cluster_management/operations.py]: done

File: csm/common/ha/cluster_management/operations.py (hunk 2 above)
Comment (shrimetta) [reply csm/common/ha/cluster_management/operations.py]: done

File: csm/common/ha/cluster_management/operations.py (hunk 3 above)
Comment (shrimetta) [reply csm/common/ha/cluster_management/operations.py]: done

File: csm/common/ha/cluster_management/operations.py (hunk 4 above)
Comment (shrimetta) [reply csm/common/ha/cluster_management/operations.py]: done

**Review by pawankumarsrivastava — APPROVED** [review]

**Pull request merged**